specs/applicationdata/ApplicationData.md

```diff
@@ -0,0 +1,428 @@
+
+This directory is ACL’d similarly to a package’s System Metadata directory (`%ProgramData%\Microsoft\Windows\AppRepository\Packages\...pkgfullname...`) e.g.
+
+```
+BUILTIN\Users:(OI)(CI)(Rc,S,RD,REA,X,RA)
+S-1-15-3-2977037414-864741429-1129033548-1928484290-1803339615-1058153653-556172075:(OI)(CI)(RX)
+BUILTIN\Users:(OI)(CI)(R)
+NT SERVICE\TrustedInstaller:(I)(F)
+NT SERVICE\TrustedInstaller:(I)(CI)(IO)(F)
+NT AUTHORITY\SYSTEM:(I)(F)
+NT AUTHORITY\SYSTEM:(I)(OI)(CI)(IO)(F)
+BUILTIN\Administrators:(I)(RX)
+BUILTIN\Administrators:(I)(OI)(CI)(IO)(GR,GE)
+S-1-15-3-1024-3635283841-2530182609-996808640-1887759898-3848208603-3313616867-983405619-2501854204:(I)(RX)
+S-1-15-3-1024-3635283841-2530182609-996808640-1887759898-3848208603-3313616867-983405619-2501854204:(I)(OI)(CI)(IO)(GR,GE)
+```
+
+### 3.4.2. Manifested Opt-In
+
+The "Machine Folder" is only available for packages that explicitly opt-in to enabling it via appxmanifest.xml
+
+```xml
+<Package>
+  ...
+  <Properties>
+    <appdata:ApplicationData>
+      <MachineFolder/>
+    </appdata:ApplicationData>
+    ...
+```
+
+### 3.4.3. Machine Path Creation/Deletion
+
+A new Undocked Deployment Extension Handler (DEH) creates the Machine path when a package in the package family is installed, if it doesn't already exist.
+
+The Machine path is removed when the last package in the package family is removed from the machine.
+
+### 3.4.4. APIs
+
+New properties and methods to access per-machine data:
+
+* ApplicationDataLocality.Machine
+* ClearMachineFolderAsync()
+* IsMachinePathSupported
+* MachineFolder
+* MachinePath
+
+**NOTE:** The per-machine APIs require the caller has the package family registered for the user -or- is running as SYSTEM.
+
+## 3.5. Unpackaged app data stores
+
+Unpackaged applications can use the ApplicationData API to access app data stores in the classic locations:
+
+|Packaged                        |Unpackaged                                                         |
+|--------------------------------|-------------------------------------------------------------------|
+|ApplicationData.LocalCachePath  |%LOCALAPPDATA%\<publisher>\<product>                               |
+|ApplicationData.LocalPath       |%LOCALAPPDATA%\<publisher>\<product>                               |
+|ApplicationData.MachinePath     |%ProgramData%\<publisher>\<product>                                |
+|ApplicationData.RoamingPath     |%APPDATA%                                                          |
+|ApplicationData.TemporaryPath   |[GetTempPath2W()](https://learn.microsoft.com/windows/win32/api/fileapi/nf-fileapi-gettemppath2w)<sup>1</sup>|
+|ApplicationData.LocalSettings   |HKCU\SOFTWARE\Classes\Local Settings\Software\<publisher>\<product>|
+|ApplicationData.LocalSettings   |HKCU\SOFTWARE\<publisher>\<product>                                |
```

Comment:
Um, `LocalSettings` is in this table twice, with two different values. And we're missing SharedLocal.

Reply:
2nd one is a typo, should be `.RoamingSettings`

In the classic Win32 world going years back, `HKCU\SOFTWARE` roams except for the `SOFTWARE\Classes\Local Settings` subtree.